cas-server-webapp-support/src/main/java/org/jasig/cas/web/report/SingleSignOnSessionsReportController.java

```diff
@@ -0,0 +1,157 @@
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.web.report;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.apache.commons.collections.Predicate;
+import org.jasig.cas.CentralAuthenticationService;
+import org.jasig.cas.ticket.Ticket;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.InternalResourceView;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Misagh Moayyed
+ */
+@Controller("singleSignOnSessionsReportController")
+@RequestMapping("/statistics/ssosessions")
+public final class SingleSignOnSessionsReportController {
+    /**
+     * The enum Sso session attribute keys.
+     */
+    private enum SsoSessionAttributeKeys {
+        AUTHENTICATED_PRINCIPAL("authenticated_principal"),
+        AUTHENTICATION_DATE("authentication_date"),
+        TICKET_GRANTING_TICKET("ticket_granting_ticket"),
+        NUMBER_OF_USES("number_of_uses");
+
+        private String attributeKey;
+
+        /**
+         * Instantiates a new Sso session attribute keys.
+         *
+         * @param attributeKey the attribute key
+         */
+        private SsoSessionAttributeKeys(final String attributeKey) {
+            this.attributeKey = attributeKey;
+        }
+
+        @Override
+        public String toString() {
+            return this.attributeKey;
+        }
+    }
+
+    private static final String ROOT_REPORT_ACTIVE_SESSIONS_KEY = "activeSsoSessions";
+
+    private static final String ROOT_REPORT_NA_KEY = "notAvailable";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SingleSignOnSessionsReportController.class);
+
+    private final ObjectMapper jsonMapper = new ObjectMapper();
+
+    private String viewPath = "/WEB-INF/view/jsp/monitoring/viewSsoSessions.jsp";
```

Comment:
It should be, yes. But we cant use the view name right now. I'll add a separate view resolver.